src/devices/Arduino/ArduinoBoard.cs

```diff
@@ -0,0 +1,177 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Analog;
+using System.Text;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Pwm;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using Iot.Device.Spi;
+
+#pragma warning disable CS1591
+
+namespace Iot.Device.Arduino
+{
+    /// <summary>
+    /// Implements an interface to an arduino board which is running Firmata.
+    /// See documentation on how to prepare your arduino board to work with this.
+    /// Note that the program will run on the PC, so you cannot disconnect the
+    /// Arduino while this driver is connected.
+    /// </summary>
+    public class ArduinoBoard : IDisposable
+    {
+        private Stream _serialPortStream;
+        private FirmataDevice _firmata;
+        private Version _firmwareVersion;
+        private string _firmwareName;
+        private List<SupportedPinConfiguration> _supportedPinConfigurations;
+
+        public ArduinoBoard(Stream serialPortStream)
+        {
+            _serialPortStream = serialPortStream;
+        }
+
+        public event Action<string, Exception> LogMessages;
+
+        public virtual void Initialize()
```

Comment:
NIT: Why is this method virtual? do you expect to subclass ArduinoBoard and have subclasses redefine the method?

Reply:
See above https://github.com/dotnet/iot/pull/1039#discussion_r407072894.